Rekor seems to have failed because the createtree job did not complete, but from what I can gather that's not true, since Rekor will not start without a tree id, and it's happily running (and you can see from the information logs that it failed to start _until_ it got the tree id).

Rekor seems to have failed because the createtree job did not complete, but from what I can gather that's not true, since Rekor will not start without a tree id, and it's happily running (and you can see from the information logs that it failed to start _until_ it got the tree id). 

My thinking is that because I added the `ttls` to jobs so they clean themselves up, the 100 seconds might be too short and the job goes away.

Here's the evidence:

This is from the failing action log.
```
 Wait for Rekor ready
  job.batch/createsecret condition met
  error: timed out waiting for the condition on jobs/createtree
  Error: Process completed with exit code 1.
```


This is the rekor pod log, notice that it's running happily, and failed to start up until the tree id was populated by the `createtree` job above:
```
Warning  Failed     7m3s (x3 over 7m4s)   kubelet            Error: couldn't find key treeID in ConfigMap rekor-system/rekor-config
```



```
--- rekor-system pods ---
NAME                                      READY   STATUS    RESTARTS   AGE
redis                                     1/1     Running   0          7m12s
rekor-00001-deployment-7bfdc55bd5-bp7f8   2/2     Running   0          7m11s
 describe pods pod/redis
 describe pods pod/rekor-00001-deployment-7bfdc55bd5-bp7f8
  Name:         rekor-00001-deployment-7bfdc55bd5-bp7f8
  Namespace:    rekor-system
  Priority:     0
  Node:         kind-worker/172.18.0.3
  Start Time:   Thu, 11 Aug 2022 06:26:40 +0000
  Labels:       app=rekor-00001
                pod-template-hash=7bfdc55bd5
                serving.knative.dev/configuration=rekor
                serving.knative.dev/configurationGeneration=1
                serving.knative.dev/configurationUID=622eba49-707b-4660-b752-47609b9445c3
                serving.knative.dev/revision=rekor-00001
                serving.knative.dev/revisionUID=16a3db6a-bf42-4723-8c05-bc63f6b59447
                serving.knative.dev/service=rekor
                serving.knative.dev/serviceUID=cba3092c-b49c-4636-9e5b-ff151f2ee82d
  Annotations:  autoscaling.knative.dev/min-scale: 1
                serving.knative.dev/creator: kubernetes-admin
  Status:       Running
  IP:           10.244.1.29
  IPs:
    IP:           10.244.1.29
  Controlled By:  ReplicaSet/rekor-00001-deployment-7bfdc55bd5
  Containers:
    rekor:
      Container ID:  containerd://80b308b6d8c2d42a4f898f19daf6210273827c3371a5f4d5bf2e258aeb550d65
      Image:         gcr.io/projectsigstore/rekor-server@sha256:516651575db19412c94d4260349a84a9c30b37b5d2635232fba669262c5cbfa6
      Image ID:      gcr.io/projectsigstore/rekor-server@sha256:516651575db19412c94d4260349a84a9c30b37b5d2635232fba669262c5cbfa6
      Port:          3000/TCP
      Host Port:     0/TCP
      Args:
        serve
        --trillian_log_server.address=log-server.trillian-system.svc
        --trillian_log_server.port=80
        --rekor_server.address=0.0.0.0
        --redis_server.address=redis.rekor-system.svc
        --redis_server.port=6379
        --enable_retrieve_api=true
        --trillian_log_server.tlog_id=$(TREE_ID)
        --log_type=prod
        --rekor_server.signer=memory
        --enable_attestation_storage=true
        --attestation_storage_bucket=file:///tmp/
      State:          Running
        Started:      Thu, 11 Aug 2022 06:27:02 +0000
      Ready:          True
      Restart Count:  0
      Environment:
        TREE_ID:          <set to the key 'treeID' of config map 'rekor-config'>  Optional: false
        PORT:             3000
        K_REVISION:       rekor-00001
        K_CONFIGURATION:  rekor
        K_SERVICE:        rekor
      Mounts:
        /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-rcq6q (ro)
    queue-proxy:
      Container ID:   containerd://a0c612f1e1e1cac8b5342099af7c070a477ab1597b84174d0f0477ee62e5f5c5
      Image:          gcr.io/knative-releases/knative.dev/serving/cmd/queue@sha256:813ea20d55b5063596cf967d1c63f51b9e34f883653957157b9b5341dfad0001
      Image ID:       gcr.io/knative-releases/knative.dev/serving/cmd/queue@sha256:813ea20d55b5063596cf967d1c63f51b9e34f883653957157b9b5341dfad0001
      Ports:          8022/TCP, 9090/TCP, 9091/TCP, 8012/TCP, 8112/TCP
      Host Ports:     0/TCP, 0/TCP, 0/TCP, 0/TCP, 0/TCP
      State:          Running
        Started:      Thu, 11 Aug 2022 06:26:47 +0000
      Ready:          True
      Restart Count:  0
      Requests:
        cpu:      25m
      Readiness:  http-get http://:8012/ delay=0s timeout=1s period=10s #success=1 #failure=3
      Environment:
        SERVING_NAMESPACE:                 rekor-system
        SERVING_SERVICE:                   rekor
        SERVING_CONFIGURATION:             rekor
        SERVING_REVISION:                  rekor-00001
        QUEUE_SERVING_PORT:                8012
        QUEUE_SERVING_TLS_PORT:            8112
        CONTAINER_CONCURRENCY:             0
        REVISION_TIMEOUT_SECONDS:          300
        SERVING_POD:                       rekor-00001-deployment-7bfdc55bd5-bp7f8 (v1:metadata.name)
        SERVING_POD_IP:                     (v1:status.podIP)
        SERVING_LOGGING_CONFIG:            
        SERVING_LOGGING_LEVEL:             
        SERVING_REQUEST_LOG_TEMPLATE:      {"httpRequest": {"requestMethod": "{{.Request.Method}}", "requestUrl": "{{js .Request.RequestURI}}", "requestSize": "{{.Request.ContentLength}}", "status": {{.Response.Code}}, "responseSize": "{{.Response.Size}}", "userAgent": "{{js .Request.UserAgent}}", "remoteIp": "{{js .Request.RemoteAddr}}", "serverIp": "{{.Revision.PodIP}}", "referer": "{{js .Request.Referer}}", "latency": "{{.Response.Latency}}s", "protocol": "{{.Request.Proto}}"}, "traceId": "{{index .Request.Header "X-B3-Traceid"}}"}
        SERVING_ENABLE_REQUEST_LOG:        false
        SERVING_REQUEST_METRICS_BACKEND:   prometheus
        TRACING_CONFIG_BACKEND:            none
        TRACING_CONFIG_ZIPKIN_ENDPOINT:    
        TRACING_CONFIG_DEBUG:              false
        TRACING_CONFIG_SAMPLE_RATE:        0.1
        USER_PORT:                         3000
        SYSTEM_NAMESPACE:                  knative-serving
        METRICS_DOMAIN:                    knative.dev/internal/serving
        SERVING_READINESS_PROBE:           {"tcpSocket":{"port":3000,"host":"127.0.0.1"},"successThreshold":1}
        ENABLE_PROFILING:                  false
        SERVING_ENABLE_PROBE_REQUEST_LOG:  false
        METRICS_COLLECTOR_ADDRESS:         
        CONCURRENCY_STATE_ENDPOINT:        
        CONCURRENCY_STATE_TOKEN_PATH:      /var/run/secrets/tokens/state-token
        HOST_IP:                            (v1:status.hostIP)
        ENABLE_HTTP2_AUTO_DETECTION:       false
      Mounts:
        /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-rcq6q (ro)
  Conditions:
    Type              Status
    Initialized       True 
    Ready             True 
    ContainersReady   True 
    PodScheduled      True 
  Volumes:
    kube-api-access-rcq6q:
      Type:                    Projected (a volume that contains injected data from multiple sources)
      TokenExpirationSeconds:  3607
      ConfigMapName:           kube-root-ca.crt
      ConfigMapOptional:       <nil>
      DownwardAPI:             true
  QoS Class:                   Burstable
  Node-Selectors:              <none>
  Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                               node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
  Events:
    Type     Reason     Age                   From               Message
    ----     ------     ----                  ----               -------
    Normal   Scheduled  7m11s                 default-scheduler  Successfully assigned rekor-system/rekor-00001-deployment-7bfdc55bd5-bp7f8 to kind-worker
    Normal   Pulling    7m10s                 kubelet            Pulling image "gcr.io/projectsigstore/rekor-server@sha256:516651575db19412c94d4260349a84a9c30b37b5d2635232fba669262c5cbfa6"
    Normal   Pulled     7m4s                  kubelet            Successfully pulled image "gcr.io/projectsigstore/rekor-server@sha256:516651575db19412c94d4260349a84a9c30b37b5d2635232fba669262c5cbfa6" in 6.60534617s
    Normal   Pulled     7m4s                  kubelet            Container image "gcr.io/knative-releases/knative.dev/serving/cmd/queue@sha256:813ea20d55b5063596cf967d1c63f51b9e34f883653957157b9b5341dfad0001" already present on machine
    Normal   Created    7m4s                  kubelet            Created container queue-proxy
    Normal   Started    7m4s                  kubelet            Started container queue-proxy
    Warning  Failed     7m3s (x3 over 7m4s)   kubelet            Error: couldn't find key treeID in ConfigMap rekor-system/rekor-config
    Warning  Unhealthy  6m50s (x4 over 7m3s)  kubelet            Readiness probe failed: Get "http://10.244.1.29:8012/": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
    Normal   Pulled     6m49s (x3 over 7m4s)  kubelet            Container image "gcr.io/projectsigstore/rekor-server@sha256:516651575db19412c94d4260349a84a9c30b37b5d2635232fba669262c5cbfa6" already present on machine
    Normal   Created    6m49s                 kubelet            Created container rekor
    Normal   Started    6m49s                 kubelet            Started container rekor
--- trillian-system pods ---

```

_Originally posted by @vaikas in https://github.com/sigstore/scaffolding/issues/285#issuecomment-1211622072_

--- a/config/ctlog/certs/300-createconfig.yaml
+++ b/config/ctlog/certs/300-createconfig.yaml
@@ -5,8 +5,8 @@ metadata:
   name: createctconfig
   namespace: ctlog-system
 spec:
-  backoffLimit: 20
-  ttlSecondsAfterFinished: 100
+  backoffLimit: 30
+  ttlSecondsAfterFinished: 600
   template:
     spec:
       serviceAccountName: createctconfig

--- a/config/ctlog/createtree/300-createtree.yaml
+++ b/config/ctlog/createtree/300-createtree.yaml
@@ -5,8 +5,8 @@ metadata:
   name: createtree
   namespace: ctlog-system
 spec:
-  backoffLimit: 20
-  ttlSecondsAfterFinished: 100
+  backoffLimit: 30
+  ttlSecondsAfterFinished: 600
   template:
     spec:
       serviceAccountName: createtree

--- a/config/fulcio/certs/300-createcerts.yaml
+++ b/config/fulcio/certs/300-createcerts.yaml
@@ -5,8 +5,8 @@ metadata:
   name: createcerts
   namespace: fulcio-system
 spec:
-  backoffLimit: 20
-  ttlSecondsAfterFinished: 100
+  backoffLimit: 30
+  ttlSecondsAfterFinished: 600
   template:
     spec:
       serviceAccountName: createcerts

--- a/config/rekor/createsecret/300-createsecret.yaml
+++ b/config/rekor/createsecret/300-createsecret.yaml
@@ -6,7 +6,7 @@ metadata:
   namespace: rekor-system
 spec:
   backoffLimit: 30
-  ttlSecondsAfterFinished: 100
+  ttlSecondsAfterFinished: 600
   template:
     spec:
       serviceAccountName: createsecret

--- a/config/rekor/createtree/300-createtree.yaml
+++ b/config/rekor/createtree/300-createtree.yaml
@@ -5,8 +5,8 @@ metadata:
   name: createtree
   namespace: rekor-system
 spec:
-  backoffLimit: 20
-  ttlSecondsAfterFinished: 100
+  backoffLimit: 30
+  ttlSecondsAfterFinished: 600
   template:
     spec:
       serviceAccountName: createtree

--- a/config/trillian/createdb/300-createdb.yaml
+++ b/config/trillian/createdb/300-createdb.yaml
@@ -4,8 +4,8 @@ metadata:
   name: createdb
   namespace: trillian-system
 spec:
-  backoffLimit: 20
-  ttlSecondsAfterFinished: 100
+  backoffLimit: 30
+  ttlSecondsAfterFinished: 600
   template:
     spec:
       serviceAccountName: createdb

--- a/config/trillian/updatetree/200-updatetree-job.yaml
+++ b/config/trillian/updatetree/200-updatetree-job.yaml
@@ -4,7 +4,7 @@ metadata:
   name: inactive-tree
   namespace: trillian-system
 spec:
-  ttlSecondsAfterFinished: 100
+  ttlSecondsAfterFinished: 600
   template:
     spec:
       restartPolicy: Never

--- a/config/tuf/createsecret/300-create-secret.yaml
+++ b/config/tuf/createsecret/300-create-secret.yaml
@@ -5,8 +5,8 @@ metadata:
   name: createsecret
   namespace: tuf-system
 spec:
-  backoffLimit: 20
-  ttlSecondsAfterFinished: 100
+  backoffLimit: 30
+  ttlSecondsAfterFinished: 600
   template:
     spec:
       serviceAccountName: createsecret

